### ISSUE TITLE
qemu-system-aarch64: Failed to put registers after init: Invalid argument

### ISSUE THREAD

**Issue by Asphaltt** [danobi/vmtest#74]
```bash
# qemu-system-aarch64 --version
QEMU emulator version 8.2.2 (Debian 1:8.2.2+ds-0ubuntu1)
Copyright (c) 2003-2023 Fabrice Bellard and the QEMU Project developers

# vmtest --version
vmtest 0.12.0

# RUST_BACKTRACE=full vmtest -k $(make -s image_name) -
=> Image.gz
===> Booting
qemu-system-aarch64: Failed to put registers after init: Invalid argument
thread 'main' panicked at /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/vmtest-0.12.0/src/qemu.rs:946:46:
called `Option::unwrap()` on a `None` value
stack backtrace:
   0:       0x5591bcf964 - std::backtrace_rs::backtrace::libunwind::trace::h95026ee104e9f9cc
                               at /build/rustc-fgzh1o/rustc-1.75.0+dfsg0ubuntu1~bpo0/library/std/src/../../backtrace/src/backtrace/libunwind.rs:104:5
   1:       0x5591bcf964 - std::backtrace_rs::backtrace::trace_unsynchronized::h2e83a8f65c091f33
                               at /build/rustc-fgzh1o/rustc-1.75.0+dfsg0ubuntu1~bpo0/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:       0x5591bcf964 - std::sys_common::backtrace::_print_fmt::h16497fad8fefa1b4
                               at /build/rustc-fgzh1o/rustc-1.75.0+dfsg0ubuntu1~bpo0/library/std/src/sys_common/backtrace.rs:67:5
   3:       0x5591bcf964 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h15f1aa5745f21db9
                               at /build/rustc-fgzh1o/rustc-1.75.0+dfsg0ubuntu1~bpo0/library/std/src/sys_common/backtrace.rs:44:22
   4:       0x5591c09268 - core::fmt::rt::Argument::fmt::hac58f985fe90592f
                               at /build/rustc-fgzh1o/rustc-1.75.0+dfsg0ubuntu1~bpo0/library/core/src/fmt/rt.rs:142:9
   5:       0x5591c09268 - core::fmt::write::h1fe2011c591e6e4e
                               at /build/rustc-fgzh1o/rustc-1.75.0+dfsg0ubuntu1~bpo0/library/core/src/fmt/mod.rs:1120:17
   6:       0x5591bc8604 - std::io::Write::write_fmt::h400e18cc2f11215d
                               at /build/rustc-fgzh1o/rustc-1.75.0+dfsg0ubuntu1~bpo0/library/std/src/io/mod.rs:1762:15
   7:       0x5591bcf790 - std::sys_common::backtrace::_print::haeddb0a5b45dc5de
                               at /build/rustc-fgzh1o/rustc-1.75.0+dfsg0ubuntu1~bpo0/library/std/src/sys_common/backtrace.rs:47:5
   8:       0x5591bcf790 - std::sys_common::backtrace::print::he52b6649afc92c60
                               at /build/rustc-fgzh1o/rustc-1.75.0+dfsg0ubuntu1~bpo0/library/std/src/sys_common/backtrace.rs:34:9
   9:       0x5591bd14c4 - std::panicking::default_hook::{{closure}}::h98c020b006f8b9f8
  10:       0x5591bd111c - std::panicking::default_hook::hfccf1de4b9195efa
                               at /build/rustc-fgzh1o/rustc-1.75.0+dfsg0ubuntu1~bpo0/library/std/src/panicking.rs:292:9
  11:       0x5591bd19ac - std::panicking::rust_panic_with_hook::h36749e053173d277
                               at /build/rustc-fgzh1o/rustc-1.75.0+dfsg0ubuntu1~bpo0/library/std/src/panicking.rs:779:13
  12:       0x5591bcfd10 - std::panicking::begin_panic_handler::{{closure}}::h586f009d8f482926
                               at /build/rustc-fgzh1o/rustc-1.75.0+dfsg0ubuntu1~bpo0/library/std/src/panicking.rs:649:13
  13:       0x5591bcfba4 - std::sys_common::backtrace::__rust_end_short_backtrace::h987e6421e225d207
                               at /build/rustc-fgzh1o/rustc-1.75.0+dfsg0ubuntu1~bpo0/library/std/src/sys_common/backtrace.rs:170:18
  14:       0x5591bd16bc - rust_begin_unwind
                               at /build/rustc-fgzh1o/rustc-1.75.0+dfsg0ubuntu1~bpo0/library/std/src/panicking.rs:645:5
  15:       0x55919ba044 - core::panicking::panic_fmt::hc28d35037bc25df7
                               at /build/rustc-fgzh1o/rustc-1.75.0+dfsg0ubuntu1~bpo0/library/core/src/panicking.rs:72:14
  16:       0x55919ba0b8 - core::panicking::panic::h26b9ce13c5456a2d
                               at /build/rustc-fgzh1o/rustc-1.75.0+dfsg0ubuntu1~bpo0/library/core/src/panicking.rs:127:5
  17:       0x5591aaa60c - vmtest::qemu::Qemu::extract_child_stderr::hfad2ec1829a4160e
  18:       0x5591aabaec - vmtest::qemu::Qemu::run::h78384d0ec39efc9c
  19:       0x5591a17be4 - vmtest::vmtest::Vmtest::run_one::h8dae7edb4b4d14a0
  20:       0x55919f509c - vmtest::ui::Ui::run::ha247175a06389633
  21:       0x55919d6374 - vmtest::main::h4e90dc869ddc2650
  22:       0x55919e4dfc - std::sys_common::backtrace::__rust_begin_short_backtrace::h1dc5b058f75aae43
  23:       0x55919e41a0 - std::rt::lang_start::{{closure}}::hfb77c5963a2eac21
  24:       0x5591bd1560 - core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &F>::call_once::hfacdcd77779a64f9
                               at /build/rustc-fgzh1o/rustc-1.75.0+dfsg0ubuntu1~bpo0/library/core/src/ops/function.rs:284:13
  25:       0x5591bd1560 - std::panicking::try::do_call::h854a658ca020e062
                               at /build/rustc-fgzh1o/rustc-1.75.0+dfsg0ubuntu1~bpo0/library/std/src/panicking.rs:552:40
  26:       0x5591bd1560 - std::panicking::try::hf1dd9fe7daac3142
                               at /build/rustc-fgzh1o/rustc-1.75.0+dfsg0ubuntu1~bpo0/library/std/src/panicking.rs:516:19
  27:       0x5591bd1560 - std::panic::catch_unwind::hf164b9d0bf40f1df
                               at /build/rustc-fgzh1o/rustc-1.75.0+dfsg0ubuntu1~bpo0/library/std/src/panic.rs:142:14
  28:       0x5591bd1560 - std::rt::lang_start_internal::{{closure}}::h7412889e1f7f6887
                               at /build/rustc-fgzh1o/rustc-1.75.0+dfsg0ubuntu1~bpo0/library/std/src/rt.rs:148:48
  29:       0x5591bd1560 - std::panicking::try::do_call::hcb118f4af2b0b28a
                               at /build/rustc-fgzh1o/rustc-1.75.0+dfsg0ubuntu1~bpo0/library/std/src/panicking.rs:552:40
  30:       0x5591bd1560 - std::panicking::try::h7a611b2a1f0efb5c
                               at /build/rustc-fgzh1o/rustc-1.75.0+dfsg0ubuntu1~bpo0/library/std/src/panicking.rs:516:19
  31:       0x5591bd3758 - std::panic::catch_unwind::h65e3c02ba6efe75f
                               at /build/rustc-fgzh1o/rustc-1.75.0+dfsg0ubuntu1~bpo0/library/std/src/panic.rs:142:14
  32:       0x5591bd3758 - std::rt::lang_start_internal::hab4a4a637ef9ccbf
                               at /build/rustc-fgzh1o/rustc-1.75.0+dfsg0ubuntu1~bpo0/library/std/src/rt.rs:148:20
  33:       0x55919e4188 - std::rt::lang_start::h0df05d12b2bfc685
  34:       0x55919d8a10 - main
  35:       0x7fa28784c4 - <unknown>
  36:       0x7fa2878598 - __libc_start_main
  37:       0x55919ba570 - _start
  38:                0x0 - <unknown>
```

I use `vmtest` to run BPF selftests on arm64.

```bash
# uname -a
Linux arm-bpf-dev 6.1.43-rockchip-rk3588 #1.0.8 SMP Sat Feb  3 00:04:38 CST 2024 aarch64 aarch64 aarch64 GNU/Linux

# lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu Oracular Oriole (development branch)
Release:	24.10
Codename:	oracular
```

**Comment by Asphaltt** [comment]
Then, after fixing it, I get the error message:

```bash
=> Image.gz
===> Booting
qemu-system-aarch64: Failed to put registers after init: Invalid argument


Caused by:
    0: QMP handshake failed
    1: Connection reset by peer (os error 104)
    2: Connection reset by peer (os error 104)
```

---

Next, it will be OK to run `vmtest` when try to run some times.

**Comment by Asphaltt** [comment]
Fixed by #72

**Comment by danobi** [comment]
I cut v0.13
